onnxruntime/core/platform/path_lib.h

```diff
@@ -220,6 +224,66 @@
   return input.substr(pos);
 }
 
+#elif defined(_AIX)
+inline OrtFileType DTToFileTypeAIX(struct stat st) {
+    switch (st.st_mode & _S_IFMT) {
+	case S_IFBLK:
+		return OrtFileType::TYPE_BLK;
+	case S_IFCHR:
+		return OrtFileType::TYPE_CHR;
+	case S_IFDIR:
+		return OrtFileType::TYPE_DIR;
+	case S_IFIFO:
+		return OrtFileType::TYPE_FIFO;
+	case S_IFLNK:
+		return OrtFileType::TYPE_LNK;
+	case S_IFREG:
+		return OrtFileType::TYPE_REG;
+	/* No Socket type */
+	default:
+		return OrtFileType::TYPE_UNKNOWN;
+	}
+}
+
+template <typename T>
+void LoopDir(const std::string& dir_name, T func) {
+  DIR* dir = opendir(dir_name.c_str());
+  struct stat stats; 
+  if (dir == nullptr) {
+    auto e = errno;
+    char buf[1024];
+    char* msg;
+#if defined(__GLIBC__) && defined(_GNU_SOURCE) && !defined(__ANDROID__)
+    msg = strerror_r(e, buf, sizeof(buf));
```

Comment:
would __ANDROID__ ever be defined on AIX?

Reply:
Hi @skottmckay, 
ANDROID not defined for AIX. I have removed that and pushed the changes to same branch. Thank you for catching it.